adafruit_led_animation/animation/sparkle.py

```diff
@@ -94,5 +94,6 @@ def draw(self):
     def after_draw(self):
         self.show()
         for pixel in self._pixels:
-            self.pixel_object[pixel] = self._half_color
-            self.pixel_object[pixel + 1] = self._dim_color
+            num_pixels = len(self.pixel_object)
```

Comment:
Maybe move this outside the loop to avoid having to check the length on each iteration? (This should be O(1) but avoiding as many calls as possible in loops is good)

Reply:
Good call. Since this value doesn't change during the course of execution I moved it up to the constructor.  Let me know if you'd prefer it back in the after_draw method.